### DISCUSSION TITLE
Timeline: Time Traveling

### DISCUSSION THREAD

**Discussion by thomasKuntermannDiva-e** [vuejs/devtools-v6#1692] [Q&A]
with the latest update, the event history disappeared from the available tabs. OK, it maybe now found below "Vuex Mutations", when selecting the new "timeline" icon. What I am now missing is the functionality to "restore a specific event state" on my Application under Test. I really need this to make a pop-up frame visible for further inspection with dev-tools. Can you pls let me know, how to use this older "restore timeline" function within the latest update?
<img width="1439" alt="Bildschirmfoto 2022-02-09 um 13 03 17" src="https://user-images.githubusercontent.com/27360136/153197539-c01ad8d9-4271-4ca5-8c24-a47ee756c09b.png">
The screenshot shows the new timeline. Yesterday i was still able (old version) to have the "revert timeline" on the selected event...

Thanks in advance and best regards

Thomas

**Comment by Akryum** [comment]
This is not implemented as the functionality of the devtools changed to a timeline which is not just about the Vuex mutations but also about any other events from components, plugins such as pinia, etc.

Let's use this thread to discuss about Time traveling and how it could look/work in the Timeline.

@tkuntermannpv if you really need Vuex time traveling, please use the [legacy version](https://devtools.vuejs.org/guide/installation.html#legacy).

**Reply by Akryum** [reply]
Note that you can also edit the Vuex state in the Vuex inspector.

**Reply by thomasKuntermannDiva-e** [reply]
thanks @Akryum Using the old version again :) it is just very handsome for me as website tester to travel through time on any objects, w/o knowing the exact state to set... for me we can close the issue here.

**Reply by Akryum** [reply]
We can keep this open to discuss around the topic as we might like to have time traveling built into the timeline.

**Reply by thomasKuntermannDiva-e** [reply]
sounds good! i would appreciate this functionality to be kept in the new version as well :)

**Reply by Akryum** [reply]
Well it's more about how to implement it as opposed as "keeping" it :cat2:

**Reply by amcsi** [reply]
I don't understand. So there is supposed to be time travel debugging, just not only involving Vuex, but component events, routing, and other things as well? Then I would expect that if I clicked on some point in time in the timeline, it would take me back there and I could fork off interaction with the web app from there. But the best that timeline gives me is that periodically it switches to showing a screenshot of that point in time, but it's only a screenshot; I can't actually go back in time to that point and look at the DOM in Chrome Dev Tools.

What is the future of this going to be then? Is there never going to be time travel debugging where I can just select a mutation and go to that point in time?

**Reply by thomasKuntermannDiva-e** [reply]
@amcsi the old version, i can work properly w/o having any auto-update issue -> https://devtools.vuejs.org/guide/installation.html#legacy

I really hope that the team will somehow find a solution to include this golden tool again into the new versions

**Reply by amcsi** [reply]
@tkuntermannpv I know, but I'm also worried about the future for when I would use VueJS 3 with Pinia whether I'll get to have Time-Travel Debugging like the good old days with VueJS 2 + Vuex, or even React + Redux.

**Reply by renatodeleao** [reply]
+1 for implementing it back, even if it does include those known limitations. Vuex Devtools time travel was one of my main arguments for preventing my team to role out hand-made stores using raw vue reactive variables, it's unbelievably useful! I actually googled where to find the time travel panel, that's how I got here.

**Reply by amcsi** [reply]
Apparently there's now a separate time travel debugging tool for Pinia:

https://colada.dev/

**Reply by isimmons** [reply]
Looks like no one has worked on colada for  a year and it has no declaration file for TS imports. Anyone find anything else?

**Reply by saadman13** [reply]
@Akryum I don't see the vuex inspector, could you please guide where it is on the devtools?

![image](https://github.com/vuejs/devtools/assets/55482126/a00853f6-6374-428d-b09b-c3acb0d5f54b)

**Reply by Akryum** [reply]
Click the 'Vuex' tab

**Reply by saadman13** [reply]
@Akryum I don't see a inspector here to update the state values:

![Screenshot 2024-02-07 at 2 29 45 PM](https://github.com/vuejs/devtools/assets/55482126/ea00babf-9399-4cbd-a111-db2e81211a92)

**Reply by KhalfaniW** [reply]
For the links here that have gone stale, change /devtools/ /devtools**-v6**/ to update the url

example 
https://github.com/vuejs/devtools/assets/55482126/ea00babf-9399-4cbd-a111-db2e81211a92

https://github.com/vuejs/devtools-v6/assets/55482126/ea00babf-9399-4cbd-a111-db2e81211a92

**Comment by gh-LZC** [comment]
这是一封自动回复邮件。已经收到您的来信，我会尽快回复。

**Comment by david1234121** [comment]
DavidMashby
